src/brod_auth_backend.erl

```diff
@@ -0,0 +1,3 @@
+-module(brod_auth_backend).
+
+-callback auth(Host :: string(), Sock :: gen_tcp:socket() | ssl:sslsocket(), Mod :: atom(), ClientId :: binary(), Timeout :: pos_integer(), SaslOpts :: term()) -> ok | term().
```

Comment:
would be better to have `{error, Reason}` returned if something wrong happens.
then `exit(Reason)` in `brod_sock`